json-rpc/src/methods.rs

```diff
@@ -111,8 +111,8 @@ async fn get_account_state(
     let currency_info = currencies_info(service, request).await?;
     let currencies: Vec<_> = currency_info
         .into_iter()
-        .map(|info| from_currency_code_string(&info.code).unwrap())
-        .collect();
+        .map(|info| from_currency_code_string(&info.code))
+        .collect::<Result<_, _>>()?;
```

Comment:
this looks like a good catch here, if somehow there were an errant code, this would result in an attack on this service...